setup.py

```diff
@@ -360,14 +360,16 @@ def run(self) -> None:
         for ext in self.extensions:
             if isinstance(ext, CMakeExtension):
                 print(f"Building CMake extension {ext.name}")
-                with tempfile.TemporaryDirectory() as build_dir:
-                    build_dir = Path(build_dir)
-                    package_path = Path(self.get_ext_fullpath(ext.name))
-                    install_dir = package_path.resolve().parent
-                    ext._build_cmake(
-                        build_dir=build_dir,
-                        install_dir=install_dir,
-                    )
+                # Set up incremental builds for CMake extensions
+                setup_dir = Path(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
Better to commit to the pathlib API rather than mixing with os.path:
```suggestion
                setup_dir = Path(__file__).resolve().parent
```